src/nvim/terminal.c

```diff
@@ -2042,23 +2021,35 @@ static void refresh_terminal(Terminal *term)
   }
   linenr_T ml_before = buf->b_ml.ml_line_count;
 
-  // refresh_ functions assume the terminal buffer is current
+  // refresh_ functions assume the terminal buffer is current. Don't call refresh_cursor here, as we
```

Comment:
> refresh_ functions assume the terminal buffer is current

Worth adding an `assert`?

Reply:
Only applies to `refresh_screen` and `refresh_scrollback` these days (due to the use of curbuf'd memline and buffer update functions). The former already has an assert; I'll add one for the latter. :+1: